### DISCUSSION TITLE
select timeout

### DISCUSSION THREAD

**Discussion by ccoffing** [jbruchon/elks#1624] [Q&A]
I see that the current select implementation does not update the timeout to the remaining value before returning.  Yet doing so is the standard Linux behavior, and might make some userspace implementations more efficient.  For example, the current libc sleep(3) has to call gettimeofday twice to discover the remaining time, and that is less efficient and less accurate than if the kernel just returned the value.

So my question is:  Is it by design or by accident that select does not update the timeout?

**Comment by ghaerr** [comment]
> Is it by design or by accident that select does not update the timeout?

I would suspect this is by accident. Looking at some of the source at https://github.com/jbruchon/elks/blob/master/elks/fs/select.c#L228 shows that updating the `fd_set` arrays had initially not been implemented as well. This mentions some discussion in #213.

We should probably update the timeout variable and consider this a bug.

> the current libc sleep(3) has to call gettimeofday twice to discover the remaining time

I'll look further into this and comment after reading `sleep`.
